src/DynamoRevit/DynamoRevit.cs

```diff
@@ -560,14 +592,24 @@ private static DynamoView InitializeCoreView(DynamoRevitCommandData commandData)
         {
             IntPtr mwHandle = commandData.Application.MainWindowHandle;
             var dynamoView = new DynamoView(RevitDynamoViewModel);
-            
+
             new WindowInteropHelper(dynamoView).Owner = mwHandle;
 
             handledCrash = false;
 
+            //Register DynamoExternalEventHandler so Revit will call our callback
+            //we requested with API access.
+
+            var appEventHandler = new DynamoExternalEventHandler(new Action(() =>
+            {
+                UpdateLibraryLayoutSpec();
+            }));
+
+            DynamoAppExternalEvent = ExternalEvent.Create(appEventHandler);
+
             dynamoView.Dispatcher.UnhandledException += Dispatcher_UnhandledException;
             dynamoView.Closed += OnDynamoViewClosed;
-            dynamoView.Loaded += (o, e) => UpdateLibraryLayoutSpec();
+            dynamoView.Loaded += (o, e) => DynamoAppExternalEvent.Raise();
 
             return dynamoView;
```

Comment:
for the first commit,
looks just move old codes to a function `LoadDynamoView`
replace it with splashScreen
call LoadDynamoView when DynamicSplashScreenReady.

but for the last commit, I don't understand the `DynamoExternalEventHandler`, not very clear why using these event and callback.

Reply:
This is because we found the regular event handler would not work in our case. We have some Revit API calls as part of these code and they could not run within regular event handler throwing out an error saying they needs to be run on Revit API context, so we were forced to switch to define the external event handler to use

Reply:
OK, thank you for the information